src/protocols/ethereum/components/TransferReview.vue

```diff
@@ -74,41 +79,61 @@ export default defineComponent({
       });
     }
 
-    async function transfer({ amount, recipient }: { amount: string; recipient: string }) {
+    async function transfer(
+      { amount, recipient, selectedAsset }
+      : { amount: string; recipient: string; selectedAsset: IAsset },
+    ) {
       const ethereumAdapter = ProtocolAdapterFactory.getAdapter(PROTOCOLS.ethereum);
-      try {
-        loading.value = true;
-        const { hash } = await ethereumAdapter.spend(
-          Number(amount),
-          recipient,
-          {
-            fromAccount: getLastActiveProtocolAccount(PROTOCOLS.ethereum),
-            maxPriorityFeePerGas: props.transferData.maxPriorityFeePerGas,
-            maxFeePerGas: props.transferData.maxFeePerGas,
-          },
-        );
-        return hash;
-      } catch (error: any) {
-        openTransactionFailedModal(error.message);
-        throw error;
-      } finally {
-        loading.value = false;
+      loading.value = true;
+
+      if (isSelectedAssetErc20) {
+        try {
```

Comment:
Maybe it would be better to have one `try...catch` for bot `if` and `else` instead of 2?

Reply:
I'm always trying to keep `try...catch` as simple as possible, but here it makes sense to merge them